zcash_primitives/src/zip32.rs

```diff
@@ -428,16 +443,30 @@ impl ExtendedFullViewingKey {
         })
     }
 
-    pub fn address(&self, j: DiversifierIndex) -> Result<(DiversifierIndex, PaymentAddress), ()> {
-        let (j, d_j) = self.dk.diversifier(j)?;
-        match self.fvk.vk.to_payment_address(d_j) {
-            Some(addr) => Ok((j, addr)),
-            None => Err(()),
-        }
+    /// Attempt to produce a payment address given the specified diversifier
+    /// index, and return None if the specified index does not produce a valid
+    /// diversifier.
+    pub fn address(&self, j: DiversifierIndex) -> Option<PaymentAddress> {
+        self.dk
+            .diversifier(j)
+            .and_then(|d_j| self.fvk.vk.to_payment_address(d_j))
```

Comment:
Codecov says this line isn't fully tested but it seems to be to me (for both branches of the `and_then`: https://github.com/zcash/librustzcash/pull/438/files#diff-65eea7ee46cbc6fdbcd35372d1907c9fe1c39f4e49e0cc760bd0662f25e0941bR610). False positive?